core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -452,38 +456,42 @@ private[spark] class MesosClusterScheduler(
    * This method takes all the possible candidates and attempt to schedule them with Mesos offers.
    * Every time a new task is scheduled, the afterLaunchCallback is called to perform post scheduled
    * logic on each task.
+   *
+   * @return tasks Remaining resources after scheduling tasks
    */
   private def scheduleTasks(
       candidates: Seq[MesosDriverDescription],
       afterLaunchCallback: (String) => Boolean,
-      currentOffers: List[ResourceOffer],
-      tasks: mutable.HashMap[OfferID, ArrayBuffer[TaskInfo]]): Unit = {
+      currentOffers: JList[ResourceOffer],
```

Comment:
Ah yes I originally changed it because we changed the MesosSchedulerUtils return type as I rebased, but realized now I don't have to. I'm fixing it now.